### PR TITLE
Upgrade default karmada-apiserver from v1.23.8 to v1.24.2

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
Signed-off-by: changzhen <changzhen5@huawei.com>

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Upgrade default `karmada-apiserver` from v1.23.8 to v1.24.2.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: artifacts/deploy/karmada-apiserver.yaml [hunk 1]
```diff
@@ -43,7 +43,6 @@ spec:
             - --etcd-keyfile=/etc/kubernetes/pki/karmada.key
             - --etcd-servers=https://etcd-client.karmada-system.svc.cluster.local:2379
             - --bind-address=0.0.0.0
-            - --insecure-port=0
```
Comment (RainbowMango) [inline artifacts/deploy/karmada-apiserver.yaml:46]: This flag has been removed from v1.24?

Comment (XiShanYongYe-Chang) [reply artifacts/deploy/karmada-apiserver.yaml:46]: Yes, it is.

image

https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md#changes-by-kind-2

Comment (wuyingjun-lucky) [reply artifacts/deploy/karmada-apiserver.yaml:46]: Should we also modify the `karmada init` stage by removing the flags ?

**Comment by RainbowMango** [comment]
CI passed, let's run one more time.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**